comp/api/api/apiimpl/observability/telemetry.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package observability
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"github.com/gorilla/mux"
+
+	"github.com/DataDog/datadog-agent/comp/core/telemetry"
+)
+
+const (
+	metricSubsystem = "api_server"
+	metricName      = "request_duration_seconds"
+	metricHelp      = "Request duration distribution by server, method, path, and status (in seconds)."
+)
+
+// TelemetryHandler is a middleware which sends a metric representing the duration of the request,
+// tagged with the given server name, response status code, path, and method.
+func TelemetryHandler(telemetry telemetry.Component, serverName string) mux.MiddlewareFunc {
+	return telemetryHandler(telemetry, serverName, clock.New())
+}
+
+func telemetryHandler(telemetry telemetry.Component, serverName string, clock clock.Clock) mux.MiddlewareFunc {
+	tags := []string{"servername", "status_code", "method", "path"}
+	var buckets []float64 // use default buckets
+	requestDuration := telemetry.NewHistogram(metricSubsystem, metricName, tags, metricHelp, buckets)
```

Comment:
It seems that we might be crating the same histogram twice. If both IPC and CMD server are enable, if that is the case I believe this would throw a panic.

Reply:
I was able to trigger the runtime panic 
<img width="1506" alt="image" src="https://github.com/DataDog/datadog-agent/assets/4672858/59ec23b7-6ec7-4f58-8137-7d774e2ab25c">

I added this to my configuration:

```
agent_ipc:
  host: "127.0.0.1"
  port: 9999
```

Reply:
Yes you are right I assumed Prometheus would resolve both histograms to the same internally but it just panics instead...
I updated the code to initialize the metric at the component level (and then pass it to each server)